packages/api/core/src/util/plugin-interface.ts

```diff
@@ -61,15 +73,15 @@ export default class PluginInterface implements IForgePluginInterface {
     });
 
     for (const plugin of this.plugins) {
-      void plugin.then((plugin) => plugin.init(dir, forgeConfig));
+      plugin.init(dir, forgeConfig);
```

Comment:
This is safe theoretically because the constructor is private and the only way to make an instance is via the static create method that ensures the property is initialized.

This is a fairly standard pattern

Reply:
@MarshallOfSound oh, I apologize for misunderstanding and not being straightforward enough. What I meant is that since the constructor is still used internally to create a base class is that within this constructor there's a `for` loop that seems to be executed fully synchronously, yet it depends on `this.plugins` that is actually initialized asynchronously… This means, this loop won't do anything if the `this.plugins` array is still empty and the async code won't execute before the synchronous (from my understanding of JS this is going to be the case almost everytime, as the next *tick* is considered at the end of the constructor… but I guess this might as well be undefined behaviour and possibly a mistake, at least if I understand the code correctly).

You probably want to place `plugin.init` somewhere where the code is still async and `this.plugins` is guaranteed to be non-empty…

Reply:
Oh I see, sorry reviewing diffs on my phone isn't super easy.

Yeah moving this init loop up into the promise then above should fix that. Missed that when writing this